stonesoup/moveable/movable.py

```diff
@@ -0,0 +1,532 @@
+# -*- coding: utf-8 -*-
+import datetime
+from abc import abstractmethod, ABC
+from functools import lru_cache
+from math import cos, sin
+from typing import Sequence, Tuple
+
+import numpy as np
+from scipy.linalg import expm
+
+from stonesoup.functions import cart2sphere, cart2pol, build_rotation_matrix, rotz
+from stonesoup.types.array import StateVector
+from stonesoup.base import Property
+from stonesoup.types.state import State, StateMutableSequence
+from stonesoup.models.transition import TransitionModel
+
+
+class Movable(StateMutableSequence, ABC):
+    states: Sequence[State] = Property(
+        doc="A list of States which enables the platform's history to be "
+            "accessed in simulators and for plotting. Initiated as a "
+            "state, for a static platform, this would usually contain its "
+            "position coordinates in the form ``[x, y, z]``. For a moving "
+            "platform it would contain position and velocity interleaved: "
+            "``[x, vx, y, vy, z, vz]``")
+    position_mapping: Sequence[int] = Property(
+        doc="Mapping between platform position and state vector. For a "
+            "position-only 3d platform this might be ``[0, 1, 2]``. For a "
+            "position and velocity platform: ``[0, 2, 4]``")
+    velocity_mapping: Sequence[int] = Property(
+        default=None,
+        doc="Mapping between platform velocity and state dims. If not "
+            "set, it will default to ``[m+1 for m in position_mapping]``")
+
+    # TODO: Determine where a platform coordinate frame should be maintained
+
+    def __init__(self, *args, **kwargs):
+        """
+        Ensure that the platform location and the sensor locations are
+        consistent at initialisation.
+        """
+        super().__init__(*args, **kwargs)
+        # Set values to defaults if not provided
+
+        if self.velocity_mapping is None:
```

Comment:
Should moveable insist on having at least one state in states when initiated? Does FixedMoveable need to have a velocity_mapping?

Reply:
`states` is already a required input. So this would be a check that `states` is not empty. Not sure whether this is a good idea or not. It might be.

A `FixedMovable` doesn't need a `velocity_mapping` (and in fact it will be ignored)

Reply:
Seems like it inherits states from StateMutableSequence even though it shouldn't. When it gets to StateMutableSequence.__init__ states = None and then it is set to an empty list. This could just be a problem on my side though.

Reply:
I have made two changes in a new commit (to be pushed later after the other comments are addressed):
1) `states` is now checked for emptyness.
2) `FixedMovable` checks that velocity mapping is not set.